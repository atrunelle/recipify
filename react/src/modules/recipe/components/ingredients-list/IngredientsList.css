.ingredients-list__list {
  padding: 16px 0 16px 0; }
