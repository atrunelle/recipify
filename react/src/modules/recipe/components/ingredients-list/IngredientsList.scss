.ingredients-list {
  &__list {
    padding: 16px 0 16px 0;
  }
}