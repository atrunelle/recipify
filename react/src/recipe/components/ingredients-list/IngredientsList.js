import React, { Component } from 'react';
// import './ingredientsList.css';


class IngredientsList extends Component {

    render() {
        if (!this.props.ingredients.healthLabels) return (<div></div>);
        const healthLabels = this.props.ingredients.healthLabels.reduce((acc, h) => {
            return `${acc} <br> ${h}`;
        })
        return(
            <div className="results">
                <h2>{this.props.title}</h2>
                <div>{this.props.ingredients.calories}</div>
                <div>{this.props.ingredients.totalWeight}</div>
                <div>{this.props.ingredients.dietLabels}</div>
                <div>{healthLabels}</div>
                {/* {this.props.results.totalDaily} */}
                {/* {this.props.results.totalNutrients} */}
                <div>{this.props.ingredients.cautions}</div>
                <div>{this.props.ingredients.glycemicIndex}</div>
                {/* {this.props.results.ingredients} */}
            </div>
        );
    }
}

export default IngredientsList;