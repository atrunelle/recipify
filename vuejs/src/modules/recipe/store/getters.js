import * as types from './types';

export default {
  [types.GET_RECIPE]: (state) => state.recipeIngredients,
  [types.GET_RECIPES]: (state) => state.recipes,
};
