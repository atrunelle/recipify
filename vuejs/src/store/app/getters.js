import * as types from './types';

export default {
  [types.GET_ALERT]: (state) => state.alert,
};
