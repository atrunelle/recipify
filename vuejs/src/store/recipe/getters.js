import nutritionService from '@/core/nutrition.service';

export default {
  totalWeight: (state) => {
    return state.ingredients.reduce((sum, ingredient) => {
      return ingredient.nutrients.totalWeight + sum;
    }, 0);
  },

  totalCalories: (state) => {
    return state.ingredients.reduce((sum, ingredient) => {
      return ingredient.nutrients.calories + sum;
    }, 0);
  },

  totalNutrients: (state, getters) => {
    return nutritionService
      .getTotalNutrients(state.ingredients, getters.totalCalories);
  },
};
