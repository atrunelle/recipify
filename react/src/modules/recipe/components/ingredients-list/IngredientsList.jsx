import React, { Component } from 'react';
import { 
  TextField, Button, Typography, Grid,
} from '@material-ui/core';
import IngredientDetails from '../ingredient-details/IngredientDetails';
import './IngredientsList.css';

class IngredientsList extends Component {
  constructor(props) {
    super(props);

    this.state = {
      recipeName: '',
    };
  }

  onRecipeNameChange = (event) => {
    this.setState({ recipeName : event.target.value });
  }

  save = () => {
    this.props.save(this.state.recipeName);
  }

  render() {
    if (!this.props.ingredients) return false;
  
    const ingredientsList = this.props.ingredients.map((ingredient, index) => 
      <IngredientDetails key={index} index={index} ingredient={ingredient} remove={this.props.remove}/>
    )
    
    return(
      <Grid 
        container
        direction="column">
        <Grid
          container
          direction="row"
          justify="space-between"
          alignItems="center">
          <Typography
            gutterBottom 
            variant="headline"
            component="h2">
            Your recipe ingredients list
          </Typography>          
          <form>
            <TextField
              label="Recipe name"
              value={this.state.recipeName}
              id="recipeNameInput"
              onChange={this.onRecipeNameChange}
            />
          </form>
        </Grid>
        <Grid
          item
          className="ingredients-list__list">
          {ingredientsList}
        </Grid>
        <Grid
          container
          spacing={8}
          justify="flex-end">
          <Grid item>
            <Button
              onClick={this.props.removeAll}
              id="removeAllButton">
              Remove all ingredients
            </Button>
          </Grid>
          <Grid item>
            <Button
              variant="raised"
              color="primary"
              id="saveButton"
              onClick={this.save}>
              Save recipe
            </Button>
          </Grid>
        </Grid>
      </Grid>
    );
  }
}

export default IngredientsList;