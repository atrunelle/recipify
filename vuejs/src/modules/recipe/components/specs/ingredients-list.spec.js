import Vuex from 'vuex';
import { shallow, createLocalVue } from '@vue/test-utils';
import IngredientsList from '../ingredients-list';
import getters from '@/modules/recipe/store/getters';

describe('Component: ingredients list', () => {
  let component;
  let localVue = createLocalVue();
  localVue.use(Vuex);

  const recipeActions = {
    removeIngredient: jest.fn(),
  };

  beforeEach(() => {
    const recipeStore = {
      namespaced: true,
      state: {
        recipeIngredients: [{
          name: 'carrot',
          nutrients: {
            calories: 100,
          },
        }, {
          name: 'spinach',
          nutrients: {
            calories: 100,
          },
        }, {
          name: 'carrot',
          nutrients: {
            calories: 100,
          },
        }],
      },
      getters,
      actions: recipeActions,
    };

    const store = new Vuex.Store({
      modules: {
        recipe: recipeStore,
      },
    });

    component = shallow(IngredientsList, {
      localVue,
      store,
    });
  });

  it('should remove ingredient', () => {
    component.find('#remove-ingredient-1').trigger('click');
    expect(recipeActions.removeIngredient).toHaveBeenCalled();
  });

  it('should match snapshot', () => {
    expect(component.html()).toMatchSnapshot();
  });
});
