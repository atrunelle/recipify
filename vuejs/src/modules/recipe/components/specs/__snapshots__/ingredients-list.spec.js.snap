// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ingredients list should match snapshot 1`] = `
<div class="v-card v-sheet theme--light">
  <div class="v-card__title justify-space-between v-card__title--primary">
    <h2 class="headline">Your recipe ingredients list</h2>
    <div class="flex md3"><input type="text" md3="" label="Recipe name" value=""></div>
  </div>
  <div role="list" class="v-list theme--light">
    <div class="v-list__group">
      <div class="v-list__group__header">
        <div role="listitem">
          <div class="v-list__tile theme--light">
            <div class="v-list__tile__content">
              <div class="v-list__tile__title">carrot cal for gr </div>
            </div>
            <div id="remove-button-0" class="v-list__tile__action"><i aria-hidden="true" class="v-icon material-icons theme--light">delete</i></div>
          </div>
        </div>
        <div class="v-list__group__header__append-icon"><i aria-hidden="true" class="v-icon material-icons theme--light">keyboard_arrow_down</i></div>
      </div>
      <div class="v-list__group__items" style="display: none;">
        <div>
          <div class="layout justify-space-between align-center row">
            <div class="flex text-xs-center">
              <h3 class="pa-2">Fat</h3>
              <p>5.00gr</p>
            </div>
            <div class="flex text-xs-center">
              <h3 class="pa-2">Fat</h3>
              <p>5.00gr</p>
            </div>
          </div>
          <div class="text-xs-center pa-2"></div>
        </div>
      </div>
    </div>
  </div>
  <div class="v-card__actions justify-end"><button type="button" class="v-btn v-btn--flat theme--light" id="remove-all-button">
      <div class="v-btn__content">
        Remove all ingredients
      </div>
    </button> <button type="button" class="v-btn theme--light primary" id="save-button">
      <div class="v-btn__content">
        Save recipe
      </div>
    </button></div>
</div>
`;
